src/Symfony/Component/Security/Guard/GuardAuthenticatorInterface.php

```diff
@@ -28,8 +29,10 @@
  * one location.
  *
  * @author Ryan Weaver <ryan@knpuniversity.com>
+ *
+ * @deprecated since version 3.4, to be removed in 4.0. Use AuthenticatorInterface instead.
  */
-interface GuardAuthenticatorInterface extends AuthenticationEntryPointInterface
+interface GuardAuthenticatorInterface extends AuthenticatorInterface
```

Comment:
I think this should be done the other way around, to still allow passing legacy type-hints.
this would mean removing the runtime notice above, but DebugClassLoader will take care or triggering it anyway

Reply:
`AuthenticatorInterface extends GuardAuthenticatorInterface`